app/controllers/mixins/ems_common_angular.rb

```diff
@@ -266,6 +266,20 @@ def ems_form_fields
                         :ssh_keypair_auth_status       => ssh_keypair_auth_status.nil? ? true : ssh_keypair_auth_status
       } if controller_name == "ems_infra"
 
+      render :json => {
+        :name                => @ems.name,
+        :emstype             => @ems.emstype,
+        :zone                => zone,
+        :provider_id         => @ems.provider_id ? @ems.provider_id : "",
+        :hostname            => @ems.hostname,
+        :default_hostname    => @ems.connection_configurations.default.endpoint.hostname,
+        :default_api_port    => @ems.connection_configurations.default.endpoint.port,
+        :provider_region     => @ems.provider_region,
+        :default_userid      => @ems.authentication_userid ? @ems.authentication_userid : "",
+        :ems_controller      => controller_name,
+        :default_auth_status => default_auth_status,
+      } if controller_name == "ems_physical_infra"
```

Comment:
Agree, there is a good scope for a refactor here, but just not in this PR.
Note that most of the code here is adhering to the existing code pattern, which has gotten crazy over the last few months when folks keep adding more and more controllers here.